runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -318,6 +318,18 @@ public Job translate(List<DataflowPackage> packages) {
 
       WorkerPool workerPool = new WorkerPool();
 
+      // If streaming engine is enabled set the proper experiments so that it is enabled on the
+      // back end as well.
```

Comment:
here you want if (streamingEngineEnabled(options))

Reply:
Currently the dependency is one way the flag enables the experiments but the experiments don't enable the flag.  streamingEngineEnabled checks all three but using it here would introduce a circular dependency.

> Adding dependency on module 'beam-runners-google-cloud-dataflow-java-fn-api-worker_main' will introduce circular dependency between modules 'beam-runners-google-cloud-dataflow-java-fn-api-worker_main' and 'beam-runners-google-cloud-dataflow-java_main'.

Reply:
Also these changes are now in https://github.com/apache/beam/pull/7693 so I'm resolving this one here and we can talk in person or on that PR